src/item_group.cpp

```diff
@@ -103,6 +103,23 @@ item Single_item_creator::create_single( const time_point &birthday, RecursionLi
     return tmp;
 }
 
+static pocket_type guess_pocket_for( const item &container, const item &payload )
```

Comment:
@KorGgenT  It throws error. You mean:
```suggestion
static item_pocket::pocket_type guess_pocket_for( const item &container, const item &payload )
```
? I'm not a C++ programmer, sorry.

Reply:
no i do not.
I don't know what error you're seeing, but a `static` function is a function that is only available in that one cpp file. what Kevin said about scope is correct; we want to keep it as small as possible. defining a static function like that also means it is not a part of the class, so you can't really use class member unless you're passing itself as an argument. making it a static function is what i thought we were gonna do when kevin mentioned scoping it basically only to item groups. it'll also discourage people from reusing the function while we're still waiting on a more complete resolution which i mentioned on an earlier comment about creating a new json member in item groups for pocket type.

Reply:
src/item_group.cpp:106:8: error: unknown type name 'pocket_type'; did you mean 'item_pocket::pocket_type'?
static pocket_type guess_pocket_for( const item &container, const item &payload )
       ^~~~~~~~~~~
       item_pocket::pocket_type
src/item_pocket.h:33:14: note: 'item_pocket::pocket_type' declared here
        enum pocket_type {
             ^
1 error generated.
make: *** [obj/tiles/item_group.o] Error 1

Reply:
argh my mistake. i apologize. i really need to wait to review things until after i've woken up fully. anyway, the static makes it good now, clang tidy won't yell you for it.